### PR TITLE
[Device management] Verify current session (PSG-722)

### LINKED ISSUE #7114: [Device management] Verify current session
#### What would you like to do?
Start verification process for unverified current session when pressing the corresponding button in main Session list and in Session overview.

#### Why would you like to do it?
Verify unverified sessions.

#### How would you like to achieve it?
Calling the verification process when pressing a Verify button for current session.

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Start the verification process when pressing the "Verify" button only for the current Session.
Bonus: Fix missing sending of verification event to other verified sessions in the devices list V1.

## Motivation and context

Closes #7114 

## Screenshots / GIFs

No UI has been changed.

## Tests

- Enable the new device management feature flag
- Sign in on a new device
- Go to Settings -> Security & Privacy -> Show all sessions (V2, WIP)
- Check the current session is displayed as not verified
- Press the "Verify" button
- Check another verified session received the request
- Cancel it
- Go to the session overview screen by pressing the current Session card
- Press the "Verify" button
- Check another verified session received the request
- Do the verification process
- Check the current session is shown as verified in the overview screen
- Press back
- Check the current session is shown as verified in the sessions list screen

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt:158]: Passing a whole Session object in a ViewEvent is a bit strange, but it looks like this is actually not used. Please remove it.

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewFragment.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewFragment.kt:117]: I understand that this Fragment is displaying only one session. So passing the device Id here is probably useless, the ViewModel already knows it.
I would do some cleanup.

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewEvent.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewEvent.kt:23]: Please remove `session`, this is not used.

**Review by bmarty — COMMENTED** [review]
Nice work, just some small cleanup needed. Thanks!

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/DevicesViewModel.kt]: You are absolutely right. It was a wrong copy-paste of the previous version. I removed it: [87c2b94](https://github.com/vector-im/element-android/pull/7139/commits/87c2b94ff91ac4f8765fe908b5a0924c595334ba)

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewEvent.kt
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewEvent.kt]: Fixed in [87c2b94](https://github.com/vector-im/element-android/pull/7139/commits/87c2b94ff91ac4f8765fe908b5a0924c595334ba)

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewFragment.kt
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewFragment.kt]: Yes removed here: [4f3df2c](https://github.com/vector-im/element-android/pull/7139/commits/4f3df2c213ba4325ab43201e702eb36200fd3009). Thanks for the review!

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt:71]: You could use the value which is in the state:
```suggestion
        if (viewState.isCurrentSession) {
```

Also there is not else block? Maybe handled in another PR? I would add a // TODO if it's the case, to not forget.
I'm scared of having a button with no effect, but maybe this is not possible (did not check).

**Review by bmarty — APPROVED** [review]
Thanks for the update. One last remark.

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt:71]: Yes, you are right but no need for a TODO as it it already implemented in the following PR #7156 which will be merged just right after this one.

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt:71]: And for the suggestion, I will do the update in the following PR to avoid blocking this one.

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt
Comment (mnaturel) [inline vector/src/main/java/im/vector/app/features/settings/devices/v2/overview/SessionOverviewViewModel.kt:49]: Remove the use case and use the field from `DeviceFullInfo` which will be available after merge of #7171

**Pull request merged**
